### ISSUE TITLE
Driver configuration system

### ISSUE THREAD

**Issue by thatoddmailbox** [foneOS/foneOS#24]
This is splitting the first item of #2 into another issue.

It should be possible to change hardware _without_ requiring a recompile AND to change pinouts also. This may have to wait on #10 first, and also requires some configuration language. It probably would be something XML-based, because pugixml is already a dependency and adding another dependency is annoying.
